src/rdfTypes/Variable.cpp

```diff
@@ -98,3 +77,27 @@ void Variable::appendEscapedWord(std::string_view word,
     }
   }
 }
+
+namespace {
+struct IsVariableVisitor {
+  using VarContext = SparqlAutomaticParser::VarContext*;
+  bool visit(VarContext) { return true; }
+  CPP_template(typename T)(
+      requires(!ad_utility::SimilarTo<T, VarContext>)) bool visit(T) {
+    return false;
+  }
+};
+}  // namespace
+
+// _____________________________________________________________________________
+bool Variable::isValidVariableName(std::string_view var) {
+  sparqlParserHelpers::ParserAndVisitorBase<IsVariableVisitor> parserAndVisitor{
+      std::string{var}};
```

Comment:
Currently the header which provides this implementation still depends on a bunch of headers that should not trigger a recompilation of `Variable.cpp` (and can easily be avoided if `ParserAndVisitor` and `ParserAndVisitorBase` are split into different files.